internal/retryhttp/client.go

```diff
@@ -133,3 +175,39 @@ func transport() http.RoundTripper {
 	}
 	return cleanhttp.DefaultPooledTransport()
 }
+
+// retryPolicy is a modified version of retryablehttp.DefaultRetryPolicy to handle
+// status codes differently.
+func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	if ctx.Err() != nil {
+		return false, ctx.Err()
+	}
+
+	if err != nil {
+		if v, ok := err.(*url.Error); ok {
+			// Don't retry if the error was due to too many redirects.
+			if redirectsErrorRe.MatchString(v.Error()) {
+				return false, nil
+			}
+
+			// Don't retry if the error was due to an invalid protocol scheme.
+			if schemeErrorRe.MatchString(v.Error()) {
+				return false, nil
+			}
+
+			// Don't retry if the error was due to TLS cert verification failure.
+			if _, ok := v.Err.(x509.UnknownAuthorityError); ok {
+				return false, nil
+			}
+		}
+
+		// The error is likely recoverable so retry.
+		return true, err
```

Comment:
This change is also lifted from the default retry policy of the retry client library.